File: app/src/main/java/com/example/volleyrvpractice/ViewClass/FloatingImageView.java
package com.example.volleyrvpractice.ViewClass;

import android.content.Context;
import android.util.AttributeSet;
import android.view.animation.Animation;
import android.view.animation.AnimationSet;
import android.view.animation.ScaleAnimation;
import android.view.animation.TranslateAnimation;

import androidx.annotation.Nullable;

public class FloatingImageView extends androidx.appcompat.widget.AppCompatImageView {

    private Animation animation1;
    private Animation animation2;
    private Animation animation3;
    private Animation animation4;

    private AnimationSet animSet1;
    private AnimationSet animSet2;


    public FloatingImageView(Context context) {
        super(context);
    }

    public FloatingImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public FloatingImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        animation1 = new TranslateAnimation(0f, 0f, 0f,-50f);
        animation1.setDuration(600);
        animation1.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                FloatingImageView.this.startAnimation(animSet2);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });
        animation2 = new TranslateAnimation(0f,0f,-50f,0f);
        animation2.setDuration(600);
        animation2.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                FloatingImageView.this.startAnimation(animSet1);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });

        animation3 = new ScaleAnimation(
                1f,0.95f,1f,0.95f,Animation.RELATIVE_TO_SELF,0.5f,Animation.RELATIVE_TO_SELF,0.5f);
        animation3.setDuration(600);

        animation4 = new ScaleAnimation(
                0.95f,1f,0.95f,1f,Animation.RELATIVE_TO_SELF,0.5f,Animation.RELATIVE_TO_SELF,0.5f);
        animation4.setDuration(600);

        animSet1 = new AnimationSet(true);
        animSet1.addAnimation(animation1);
        animSet1.addAnimation(animation4);
        animSet2 = new AnimationSet(true);
        animSet2.addAnimation(animation2);
        animSet2.addAnimation(animation3);
        FloatingImageView.this.startAnimation(animSet1);
    }
}

File: app/src/main/java/com/jasonstudio/cookbook2/view/custom/FloatingImageView.kt
package com.jasonstudio.cookbook2.view.custom

import android.content.Context
import android.util.AttributeSet
import android.view.animation.Animation
import androidx.appcompat.widget.AppCompatImageView
import android.view.animation.AnimationSet
import android.view.animation.TranslateAnimation
import android.view.animation.ScaleAnimation

class FloatingImageView : AppCompatImageView {
    private lateinit var animation1: Animation
    private lateinit var animation2: Animation
    private lateinit var animation3: Animation
    private lateinit var animation4: Animation
    private lateinit var animSet1: AnimationSet
    private lateinit var animSet2: AnimationSet

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        animation1 = TranslateAnimation(0f, 0f, 0f, -50f)
        animation1.setDuration(600)
        animation1.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {
                startAnimation(animSet2)
            }

            override fun onAnimationRepeat(animation: Animation) {}
        })
        animation2 = TranslateAnimation(0f, 0f, -50f, 0f)
        animation2.setDuration(600)
        animation2.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {
                startAnimation(animSet1)
            }

            override fun onAnimationRepeat(animation: Animation) {}
        })
        animation3 = ScaleAnimation(
            1f, 0.95f, 1f, 0.95f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f
        )
        animation3.setDuration(600)
        animation4 = ScaleAnimation(
            0.95f, 1f, 0.95f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f
        )
        animation4.setDuration(600)
        animSet1 = AnimationSet(true)
        animSet1!!.addAnimation(animation1)
        animSet1!!.addAnimation(animation4)
        animSet2 = AnimationSet(true)
        animSet2!!.addAnimation(animation2)
        animSet2!!.addAnimation(animation3)
        startAnimation(animSet1)
    }
}